src/test/common/router/routablePath.ts

```diff
@@ -12,6 +12,30 @@ describe('RoutablePath', () => {
     })
   })
 
+  describe('evaluating uri', () => {
+    it('should fail when substitutions is not provided', () => {
+      [undefined, {}].forEach(invalidValue => {
+        expect(() => new RoutablePath('/case/:externalId/payment/:payment-type').evaluateUri(invalidValue))
+          .to.throw(Error, 'Path parameter substitutions are required')
+      })
+    })
+
+    it('should fail when not all path parameter placeholders has been replaced', () => {
+      expect(() => new RoutablePath('/case/:id/payment/:payment-type').evaluateUri({ id: '999' }))
+        .to.throw(Error, 'At least one path parameter substitution is missing')
+    })
+
+    it('should replace all path parameter placeholders', () => {
+      expect(new RoutablePath('/case/:id/payment/:payment-type').evaluateUri({ id: '999', 'payment-type': 'card' }))
+        .to.be.equal('/case/999/payment/card')
+    })
+
+    it('should ignore substitutions which are not defined as path parameter placeholders', () => {
+      expect(new RoutablePath('/case/:id').evaluateUri({ id: '999', foo: 'bar' }))
```

Comment:
Why don't we throw exception for unknown params given?

Reply:
I had this though but gave up to keep it simpler. But generally it would be good to be stricter.

Reply:
is it more complicated than looking for `/:foo/` if so leave it,
otherwise it would be nice to have